### ISSUE TITLE
Turning off `allowNormalSpawning` also disables spawners and spawn eggs

### ISSUE THREAD

**Issue by Zarainia** [Darkhax-Minecraft/BadMobs#63]
For example, if I have:

```toml
#Spawning options for minecraft:zombie
[minecraft.zombie]
  #Should the entity be allowed to spawn normally?
  allowNormalSpawning = false
  #Should spawners be able to spawn the entity?
  allowSpawners = true
  #Should spawn eggs be able to spawn the entity?
  allowSpawnEggs = true
  #Should the entity spawn via mob conversion? i.e. villager -> zombie
  allowConversions = true
  #When enabled the entity type will be aggresively removed from worlds. This will bypass all other options.
  removeAggressively = false
```

and use a spawn egg, I hear the zombie noise, but it never spawns. Not sure what could be happening; I looked at the code and it makes sense to me. 

Minecraft version: 1.18.2
Forge version: 40.1.54
Mod version: BadMobs-1.18.2-13.01.jar

**Comment by JENOVAAbsolute** [comment]
Can confirm this as well.

**Comment by Ocpkayp6662** [comment]
yeah this issue happens to me too pls fix it when you can (mine is 1.19.2)

**Comment by Jsnbrown1989** [comment]
Can also confirm on 1.16.5

**Comment by MattMcL1221** [comment]
Just spent 2 hours setting up this mod only for it not to work. Also using 1.16.5

PLEASE PLEASE FIX THIS MOD. This is the only issue stopping me from making a high quality mod pack :(

**Comment by Zarainia** [comment]
> Just spent 2 hours setting up this mod only for it not to work. Also using 1.16.5
> 
> PLEASE PLEASE FIX THIS MOD. This is the only issue stopping me from making a high quality mod pack :(

You could try https://www.curseforge.com/minecraft/mc-mods/in-control for disabling normal spawns only. The configuration is a little more complicated, but it works for me.

**Comment by Darkhax** [comment]
This issue stems from a bug in Forge. They're trying to fix it in 1.19.4, so hopefully we can finally fix this issue.

**Comment by Darkhax** [comment]
Forge was able to fix this issue on 1.19.4 and the mod appears to be working as intended now. Unfortunately older versions will not be receiving this bug fix.
